### PR TITLE
Resolve `NullSafeMutableLiveData` Lint Errors

### PR DESCRIPTION
**Pull request by ParaskP7**
Related to #16632

This PR deals with the `NullSafeMutableLiveData` Lint errors as they suddenly started appearing within the [Navigation Component 2.4.2](https://github.com/wordpress-mobile/WordPress-Android/pull/16632) update PR.

Some of these `NullSafeMutableLiveData` were fixed, while others suppressed. You will find all the details you need in the commit messages themselves as to how a specific `NullSafeMutableLiveData` got resolved. As such, I recommend reviewing this PR commit-by-commit.

PS: You will find more info about the decision behind this PR within [this conversation](https://github.com/wordpress-mobile/WordPress-Android/pull/16632#discussion_r881619708).

## To test:

All the changes were pretty straightforward and as such don't necessary require much testing. However, I would anyway recommend a quick smoke testing on the below screens and/or functionalities:
- Site Creation
- Stats
- Stats (Tabs)
- Selected Date (from Stats)
- Posts
- Posts (Search)
- Pages
- Pages (Search)
- Publish Settings (from Posts)
- Media Picker (from Post)
- Photo Picker (from Post)
- Register Domain (from Menu)
- Plans (from Menu)
- Plugins (from Menu)
- Categories (from Site Settings)
- What's New Announcement

## Regression Notes

1. Potential unintended areas of impact

`N/A`

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

`N/A`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ParaskP7** [comment]
👋 @oguzkocer @ashiagr !

FYI: Just in case you missed [my previous ping](https://github.com/wordpress-mobile/WordPress-Android/pull/16632#discussion_r883648168), this PR is now ready for review. 🎉 

I updated lifecycle to `2.4.1` and `resolved/suppressed` those 10 new `NullSafeMutableLiveData` lint errors, which did the trick and now the PR is 🟢 on CI.

Please take a look at is and let me know how else I can help you. 🙏

File: build.gradle [hunk 1]
```diff
@@ -18,13 +18,12 @@ ext {
...
     androidxArchCoreVersion = '2.0.0'
     fragmentVersion = '1.2.4'
-    lifecycleVersion = '2.4.1'
     constraintLayoutVersion = '1.1.3'
     materialVersion = '1.2.1'
     preferenceVersion = '1.1.0'
     swipeToRefresh = '1.1.0'
     uCropVersion = '2.2.4'
-    lifecycleVersion = '2.2.0'
+    lifecycleVersion = '2.4.1'
```
Comment (ashiagr) [inline build.gradle:26]: 🥇 for this! 👏

**Review by ashiagr — APPROVED** [review]
👋 @ParaskP7 

I went through [this discussion](https://github.com/wordpress-mobile/WordPress-Android/pull/16632#discussion_r881619708), and am really happy to see `lifecycle` updated to `2.4.1` with CI passing.

I was able to follow the changes by reading commit messages. I experienced no crashes while smoke testing mentioned areas. Overall I'm confident of approving the changes. 

As always, thank you for your thorough work! 🙏 

Feel free to merge when you're ready.

**Comment by ParaskP7** [comment]
1. @ashiagr Thank you so much for the review and for giving us the 🟢 light with these changes! 🥇 
2. @oguzkocer I am letting you merge this when and if you feel okay about it too so that we could then progress #16632 and ultimately #15314 .
 
PS: As part of this PR review and test we should be able to merged #16632 straight to `trunk` or we could just close it and point this PR to `trunk` directly. 🎯

**Comment by AliSoftware** [comment]
I'm moving this PR's milestone to `20.1` as I'm doing the code freeze for `20.0`.

> @oguzkocer I am letting you merge this when and if you feel okay about it too so that we could then progress https://github.com/wordpress-mobile/WordPress-Android/pull/16632 and ultimately https://github.com/wordpress-mobile/WordPress-Android/pull/15314 .

I was tempted to merge this as I saw this was approved, but given this comment + since the troubles Petros had on Friday's release finalization domino'd into [postponing the finalization and thus the code freeze today](https://github.com/wordpress-mobile/WordPress-Android/pull/16661), I'm now too late in my day to wait any further to give a chance for this to land before I start the freeze; but feel free to rebase to the release branch if you feel it's necessary)

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

> I'm moving this PR's milestone to 20.1 as I'm doing the code freeze for 20.0.

Thank you for moving this to the new `20.1` milestone, this is correct.

> I was tempted to merge this as I saw this was approved, but given this comment + since the troubles Petros had on Friday's release finalization domino'd into https://github.com/wordpress-mobile/WordPress-Android/pull/16661, I'm now too late in my day to wait any further to give a chance for this to land before I start the freeze; but feel free to rebase to the release branch if you feel it's necessary)

Thanks for the communication and for NOT merging this. This indeed doesn't and wouldn't need to be included and rebased to the release branch, so all good from our side.

**Comment by oguzkocer** [comment]
Thank you for the PR @ParaskP7 and thank you for the review @ashiagr 🙇‍♂️ I did a very quick look at it and it all makes sense to me. Note that I am not adding an explicit approval because I didn't do a proper review as I have a lot to catch up on and I don't want to block a PR that's already approved.

I also took a look at the [release notes for `lifecycle`](https://developer.android.com/jetpack/androidx/releases/lifecycle) and as far as I can tell there are only a few minor behavioural changes, so it seems like a "safe" update, so 👍 for that as well.

**Pull request merged**